app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -54,6 +54,43 @@ pub struct MissingThisOnMethod(pub String);
 
 
 
+// =====================
+// === QualifiedName ===
+// =====================
+
+im_string_newtype! {
+    /// A single segment of a [`QualifiedName`] of an [`Entry`].
+    QualifiedNameSegment
+}
+
+/// A fully qualified name of an [`Entry`].
+///
+/// A string representation of a fully qualified name can be built by joining the
+/// [`QualifiedName::segments`] using an [`ast::opr::predefined::ACCESS`] character.
+#[derive(Debug, Default, Clone, PartialEq)]
+#[allow(missing_docs)]
+pub struct QualifiedName {
+    pub segments: Vec<QualifiedNameSegment>,
+}
+
+impl<'a> FromIterator<&'a str> for QualifiedName {
+    fn from_iter<T>(iter: T) -> Self
+    where T: IntoIterator<Item = &'a str> {
+        let segments = iter.into_iter().map(|s| s.into()).collect();
+        Self { segments }
+    }
+}
```

Comment:
This is OK but it doesnt allow for efficient usage. You are clonning strings here. Write also `FromIterator` that does not requrie clonning, like `FromIterator<String>`.

Reply:
`QualifiedNameSegment` is a newtype for `ImString` and has both `From<&str>` and `From<String>` implemented. Do I understand correctly that we can just generalize over `Into<QualifiedNameSegment>` here instead, and this way get a non-cloning behavior for `String` iterators? If not, I will change this as you suggest.